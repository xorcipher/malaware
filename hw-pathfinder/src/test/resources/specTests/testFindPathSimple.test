# Find a simple path with Dijkstra's algorithm when
# the path found is much longer than multiple alternatives,
# but still the lowest cost path.

CreateGraph simpleGraph

AddNode simpleGraph A
AddNode simpleGraph B
AddNode simpleGraph C
AddNode simpleGraph D
AddNode simpleGraph E
AddNode simpleGraph F
AddNode simpleGraph G
AddNode simpleGraph H
AddNode simpleGraph I
AddNode simpleGraph J

AddEdge simpleGraph A B 10.0
AddEdge simpleGraph A C 1.0
AddEdge simpleGraph C D 1.0
AddEdge simpleGraph D B 1.0
AddEdge simpleGraph A G 0.1
AddEdge simpleGraph G F 0.1
AddEdge simpleGraph F E 0.1
AddEdge simpleGraph E H 0.1
AddEdge simpleGraph H I 0.1
AddEdge simpleGraph I J 0.1
AddEdge simpleGraph J B 0.1

FindPath simpleGraph A B
