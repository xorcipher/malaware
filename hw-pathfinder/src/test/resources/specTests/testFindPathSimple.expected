# Find a simple path with Dijkstra's algorithm when
# the path found is much longer than multiple alternatives,
# but still the lowest cost path.

created graph simpleGraph

added node A to simpleGraph
added node B to simpleGraph
added node C to simpleGraph
added node D to simpleGraph
added node E to simpleGraph
added node F to simpleGraph
added node G to simpleGraph
added node H to simpleGraph
added node I to simpleGraph
added node J to simpleGraph

added edge 10.000 from A to B in simpleGraph
added edge 1.000 from A to C in simpleGraph
added edge 1.000 from C to D in simpleGraph
added edge 1.000 from D to B in simpleGraph
added edge 0.100 from A to G in simpleGraph
added edge 0.100 from G to F in simpleGraph
added edge 0.100 from F to E in simpleGraph
added edge 0.100 from E to H in simpleGraph
added edge 0.100 from H to I in simpleGraph
added edge 0.100 from I to J in simpleGraph
added edge 0.100 from J to B in simpleGraph

path from A to B:
A to G with weight 0.100
G to F with weight 0.100
F to E with weight 0.100
E to H with weight 0.100
H to I with weight 0.100
I to J with weight 0.100
J to B with weight 0.100
total cost: 0.700